source/math/FloatComplex.ooc

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright (c) 2011-2012 Anders Frisk
+// Copyright (c) 2012-2015 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import math
+import text/StringTokenizer
+import structs/ArrayList
+
+FloatComplex: cover {
+	real, imaginary: Float
+	init: func@ (=real, =imaginary)
+	conjugate ::= FloatComplex new(this real, - this imaginary)
+	absoluteValue ::= (this real pow(2) + this imaginary pow(2)) sqrt()
+	operator + (other: This) -> This { This new(this real + other real, this imaginary + other imaginary) }
+	operator - (other: This) -> This { This new(this real - other real, this imaginary - other imaginary) }
+	operator - -> This { This new(-this real, -this imaginary) }
+	operator * (other: Float) -> This { This new(other * this real, other * this imaginary) }
+	operator * (other: This) -> This { This new(this real * other real - this imaginary * other imaginary, this real * other imaginary + this imaginary * other real) }
+	operator / (other: Float) -> This { This new(this real / other, this imaginary / other) }
+	operator / (other: This) -> This { (this * other conjugate) / (other absoluteValue pow(2)) }
+	operator == (other: This) -> Bool { this real == other real && this imaginary == other imaginary }
+	operator != (other: This) -> Bool { !(this == other) }
+	toString: func -> String {
+		this real toString() >> (this imaginary > 0 ? " +" : " ") & this imaginary toString() >> "i"
+	}
+	parse: static func (input: String) -> This {
+    realResult, imaginaryResult: Float
+    minus := input contains?('-')
+    array := input split(minus ? '-' : '+')
+    realResult = array[0] toFloat()
+    imaginaryString := array[1] trimRight('i')
+    imaginaryResult = minus ? -imaginaryString toFloat() : imaginaryString toFloat()
+    imaginaryString free()
+    array free()
+    This new (realResult, imaginaryResult)
+	}
+	exponential: func -> This {
+		(this real) exp() * This new((this imaginary) cos(), (this imaginary) sin())
+	}
+	logarithm: func -> This {
+		This new(this absoluteValue log(), atan2(this imaginary, this real))
+	}
+	rootOfUnity: func ~one (n: Int) -> This {
+		this rootOfUnity(n, 1)
+	}
+	rootOfUnity: func ~two (n: Int, k: Int) -> This {
+		This new(0, 2 * k * PI / n)
+	}
```

Comment:
You can reduce this to one function with parameter defaults:
```ooc
rootOfUnity: func (n: Int, k := 1) -> This {
```
`k` will then default to 1 if `rootOfUnity` is called with only one parameter, but you can also call `rootOfUnity` with two parameters. This means you can also remove the suffix, since there's only one function named `rootOfUnity`.